app/src/main/java/com/github/swent/echo/compose/components/TopAppBar.kt

```diff
@@ -55,19 +57,28 @@ fun TopAppBar(
                 )
             }
         },
-        actions = { // mode switch icon
-            IconButton(
-                onClick = { switchMode() },
-                modifier = Modifier.testTag("list_map_mode_button")
-            ) {
+        actions = { // search reset button and list/map mode switch button
+            if (searchMode) {
+                IconButton(
+                    onClick = resetSearch,
+                    modifier = Modifier.testTag("search_reset_button")
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.search_off),
+                        contentDescription = "Search reset icon to reset the search field"
+                    )
+                }
+            }
+            IconButton(onClick = switchMode, modifier = Modifier.testTag("list_map_mode_button")) {
                 Icon(
                     painter =
                         if (mode == MapOrListMode.MAP) {
                             rememberVectorPainter(image = Icons.Filled.List)
                         } else {
                             painterResource(id = R.drawable.map_icon)
                         },
-                    contentDescription = "Search icon to access the search screen"
+                    contentDescription =
+                        "list/map mode switch icon to switch between list and map mode"
```

Comment:
+ve point: contentDescription made more apt so that it's easier to what its function is.